functions.php

```diff
@@ -527,10 +532,75 @@ public function register_p4_page_type_taxonomy() {
 		}
 	}
 
+	/**
+	 * Save custom taxonomy for planet4 post types
+	 */
+	public function p4_save_post_type( $post_id ) {
+		// some of these checks might be redundant, but they're all nicely
+		// separated and easy to delete so I'll leave them for now.
+		// ignore autosave
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+		// check nonce
+		if ( ! isset( $_POST['p4-page-type-nonce'] ) || ! wp_verify_nonce( $_POST['p4-page-type-nonce'], basename( __FILE__ ) ) ) {
+			return;
+		}
+		// check user's capabilities
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+		// make sure there's input
+		if ( ! isset( $_POST['p4-page-type'] ) ) { // Input var okay.
+			return;
+		}
+		// make sure the term exists
+		if ( ! $selected = get_term_by( 'slug', sanitize_text_field( $_POST['p4-page-type'] ), 'p4-page-type' ) ) { // Input var okay.
+			return;
+		}
+		// make sure it's not an error
+		if ( is_wp_error( $selected ) ) {
+			return $post_id;
+		}
+		// save post type
+		wp_set_post_terms( $post_id, $selected->slug, 'p4-page-type', $append = false );
+
+		return;
+	}
+
+	/**
+	 * Add a dropdown to choose planet4 post type.
+	 */
+	public function p4_metabox_markup( $object ) {
+		get_post_meta( $object->ID );
+		$current = - 1;
+		if ( $current_term = get_the_terms( $object, 'p4-page-type' ) ) {
+			if ( ! is_wp_error( $current_term ) ) {
+				$current = $current_term[0]->slug;
+			}
+		}
+		$terms = get_terms( 'p4-page-type', [ 'hide_empty' => false ] );
+		wp_nonce_field( basename( __FILE__ ), 'p4-page-type-nonce' );
+		?>
+        <div>
+        <select name="p4-page-type"><?
+			foreach ( $terms as $term ) {
```

Comment:
When breaking up a logic block (`if`, `foreach`, etc.) with HTML output we prefer to use the 

```

Reply:
Changed foreach to alternative syntax.